lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -714,13 +767,12 @@ namespace Js
 #ifdef INTL_ICU
     typedef const char * (*GetAvailableLocaleFunc)(int);
     typedef int (*CountAvailableLocaleFunc)(void);
-    static bool findLocale(const char *langtag, CountAvailableLocaleFunc countAvailable, GetAvailableLocaleFunc getAvailable)
+    static bool findLocale(JavascriptString *langtag, CountAvailableLocaleFunc countAvailable, GetAvailableLocaleFunc getAvailable)
     {
-        UErrorCode status = U_ZERO_ERROR;
         char localeID[ULOC_FULLNAME_CAPACITY] = { 0 };
-        int32_t length = 0;
-        uloc_forLanguageTag(langtag, localeID, ULOC_FULLNAME_CAPACITY, &length, &status);
-        ICU_ASSERT(status, length > 0);
+        BCP47_TO_ICU(langtag->GetSz(), langtag->GetLength(), localeID, ULOC_FULLNAME_CAPACITY);
+
+        // TODO(jahorto): can we binary search this instead?
```

Comment:
>// TODO(jahorto): can we binary search this instead? [](start = 8, length = 52)

+1 -- can we investigate whether this list comes in sorted order? It doesn't look like we're tied to any iterator behavior so we should have random access at this point.